src/components/student-dashboard/events-panel.cjsx

```diff
@@ -0,0 +1,37 @@
+React = require 'react'
+BS    = require 'react-bootstrap'
+ReadingRow      = require './reading-row'
+HomeworkRow     = require './homework-row'
+EventRow        = require './event-row'
+UnknownEventRow = require './unknown-event-row'
+_ = require 'underscore'
+
+module.exports = React.createClass
+  displayName: 'EventsPanel'
+
+  propTypes:
+    events:   React.PropTypes.array.isRequired
+    courseId: React.PropTypes.any.isRequired
+    startAt:  React.PropTypes.object
+    endAt:    React.PropTypes.object
+    limit:    React.PropTypes.number
+    title:    React.PropTypes.string
+    className: React.PropTypes.string
+
+  renderTitle: ->
+    if @props.title
+      <h3>{@props.title}</h3>
+    else
+      <span>{@props.startAt.format("MMMM Do")} - {@props.endAt.format("MMMM Do")}</span>
+
+  renderEvent: (event) ->
+    switch event.type
+      when 'reading'  then <ReadingRow  courseId=@props.courseId key={event.id} event={event}/>
+      when 'homework' then <HomeworkRow courseId=@props.courseId key={event.id} event={event}/>
+      else
+        <UnknownEventRow courseId=@props.courseId key={event.id} event={event}/>
```

Comment:
Should this throw an `new Error("Bug: unknown event #{event.type}")`? I'm hoping that all `throw new` code will eventually `POST` to the server so we can fix the code.